### PR TITLE
hfe: Reuse stk_per_rev from previous track for HFEv3

### PR DESCRIPTION
**Pull request by ejona86**

### CODE REVIEW

File: src/image/hfe.c [hunk 1]
```diff
@@ -121,14 +124,20 @@ static bool_t hfe_open(struct image *im)
 static void hfe_seek_track(struct image *im, uint16_t track)
 {
     struct track_header thdr;
+    uint16_t old_len;
 
     F_lseek(&im->fp, im->hfe.tlut_base*512 + (track/2)*4);
     F_read(&im->fp, &thdr, sizeof(thdr), NULL);
 
     im->hfe.trk_off = le16toh(thdr.offset);
+    old_len = im->hfe.trk_len;
```
Comment (keirf) [inline src/image/hfe.c:133]: I guess this is 0 on the first seek after mounting the image, so this should just work and cause stk_per_rev to be estimated on first seek?

**Comment by keirf** [comment]
Would we be better snapping stk_per_rev to 300 or 360 rpm if we can? And/or even more aggressively caching stk_per_rev across tracks? (Since most disks are constant RPM except Mac GCR, right?)

The problem with ever computing stk_per_rev here for HFEv3 is you know it's basically wrong. That said, what you have in this patch obviously works pretty well in your testing and is a clear improvement on the existing code.

**Comment by ejona86** [comment]
Snapping to 300 or 360 is interesting. But I think it really doesn't matter. This is used until we have one full revolution, at which we get the real value at which point it'd be stable (modulo us dealing with skipbits for writes). Well, the first rotation doesn't have to be perfect as the host is waiting for the floppy to spin up. So anything close is probably good.

I do think it'd be fair to remove the "are the tracks similar" check and _only_ set it on the first seek. If the track sizes are different, it is more likely the bitcell size changed. I'll make that change.

**Comment by keirf** [comment]
Agreed that's a good idea. What do you think about initialising `tracklen_ticks` at the same time? This could be useful for future patches, including PR #822 ?

**Comment by ejona86** [comment]
I've made the change, but I've not yet fully tested it. My control test is busted and I need to investigate more.

**Comment by keirf** [comment]
It LGTM so I squashed it all together and applied to master. If there is any fallout we can follow up.

**Pull request closed without merging**